game-core/src/main/java/games/strategy/engine/framework/startup/ui/pbem/PbemSetupPanel.java

```diff
@@ -184,11 +200,27 @@ public boolean canGameStart() {
   @Override
   public void postStartGame() {
     final GameData data = gameSelectorModel.getGameData();
+
+    Preconditions.checkNotNull(
+        data,
+        "Game Data must not be null when starting a game, "
+            + "this error indicates a programming bug that allowed for the start game button to be "
+            + "enabled without first valid game data being loaded. ");
     if (diceServerEditor.areFieldsValid()) {
       diceServerEditor.applyToGameProperties(data.getProperties());
     }
-    if (forumPosterEditor.areFieldsValid()) {
-      forumPosterEditor.applyToGameProperties(data.getProperties());
+    if (forumPosterEditorViewModel.areFieldsValid()) {
+      final GameProperties properties = data.getProperties();
```

Comment:
Instead of having the view (or view model) set game properties, which is mixing of domains, we instead get data from the view model and this class is now responsible for binding game properties to the properties we have read from UI. This way the UI model is not as tied to the model.